pkg/util/sketch/heavykeeper.go

```diff
@@ -0,0 +1,220 @@
+package sketch
+
+import (
+	"container/heap"
+	"github.com/axiomhq/hyperloglog"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"math"
+	"math/rand"
+	"sort"
+)
+
+type heavyKeeper struct {
```

Comment:
Are we going to use heavy keeper?

Reply:
Not unless we can come up with a better, more accurate merging mechanism.

At the moment our current issue is that merging of both CMS (with conservative updates) and HK result in a loss of accuracy. It's not end of the world bad, we might go from 99/100 topk correct to 90/100, slightly less for HK.

Reply:
Hm. Should we check the code into main then? I'm on the fence

Reply:
For sure, I'm fine to remove it (and sketchbf), I just wanted everyone whose interested to have the opportunity to see the other structure types.